.github/workflows/gh-ci.yaml

```diff
@@ -108,7 +108,52 @@ jobs:
         file: ./coverage.xml
         verbose: True
         # name: codecov-${{ matrix.os }}-py${{ matrix.python-version }}
-        
+  
+  main_tests_cuda:
+    name: CUDA CI (py-${{ matrix.python-version }})
+    runs-on: ubuntu-latest
+    strategy:
+        fail-fast: false
+        matrix:
+          python-version: ["3.9", "3.10"]
```

Comment:
Selfish curiosity - is this action meant to be run on GPUs? Or is it just trying to get at something at the API level that doesn't need hardware?

Reply:
Yes, I think this needs GPUs to run, so this CI might not go anywhere since [GitHub doesn't have widely available GPU runners yet](https://resources.github.com/devops/accelerate-your-cicd-with-arm-and-gpu-runners-in-github-actions/) -- this test might have to be for local testing only. Although I'm having difficulty reproducing this on the same machine I saw it on in the first place...